akka-docs-dev/rst/java/code/docs/http/javadsl/HandlerExampleSpec.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.http.javadsl;
+
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.server.*;
+import akka.http.javadsl.server.values.Parameters;
+import akka.http.javadsl.server.values.PathMatchers;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.http.javadsl.testkit.TestRoute;
+import org.junit.Test;
+
+public class HandlerExampleSpec extends JUnitRouteTest {
+    @Test
+    public void testSimpleHandler() {
+        //#simple-handler-example-full
+        class TestHandler extends akka.http.javadsl.server.AllDirectives {
+            //#simple-handler
+            Handler handler = new Handler() {
```

Comment:
Wouldn't that be an argument for all SAM classes? I think it somewhat makes sense when the parameters are fixed to provide at least an alias. We could change the classes to extend from the matching FunctionX interfaces, though, and change the `handleWith` signatures to take matching `FunctionN`, instead. WDYT?